### PR TITLE
Wrap `PreviewModels` in `DEBUG` directives.

### PR DESCRIPTION
**Pull request by markst**
In our case we have our `PreviewProvider`'s wrapped in compiler directives.
This is due to `MediaMock` being also wrapped in compiler directives and so forth.

```swift
#if DEBUG
import SwiftUI
import Common
import Prefire

internal class BannerModuleViewPreview: PreviewProvider, PrefireProvider {
    static var previews: some View {
        UIElementPreview {
            ViewRepresentable(
                view: BannerModuleView(
                    media: MediaMock.bannerMock()
                )
            )
            .snapshot(delay: 0.1, precision: 1)
            .frame(width: 375, height: 400)
            .padding(16)
            .previewLayout(.sizeThatFits)
        }
    }
}
#endif
```

### CODE REVIEW

**Comment by markst** [comment]
I wonder if there is a better solution?
It would be much better to be able to only perform plugin `.when(configuration: .debug)`

**Comment by BarredEwe** [comment]
I think we can wrap only the part with the call. This will save work in places where `#if DEBUG` is not used.

```swift
public static var models: [PreviewModel] = {
    var views: [PreviewModel] = []
    #if DEBUG
    {% for type in types.types where type.implements.PrefireProvider or type.based.PrefireProvider or type|annotated:"PrefireProvider" %}
    views.append(contentsOf: createModel(for: {{ type.name }}.self, name: "{{ type.name | replace:"_Previews","" | replace:"_Preview","" }}"))
    {% endfor %}
    #endif
    return views.sorted(by: { $0.name > $1.name || $0.story ?? "" > $1.story ?? "" })
}()


```

**Comment by markst** [comment]
Not sure whether we can configure Sourcery to identify if a `PrefireProvider` is within a DEBUG compiler directive.

I suppose we may not want to wrap in DEBUG since some people may wish to create a distribution build with Playbook?

**Comment by markst** [comment]
We could introduce a new compiler flag such as `NO_PLAYBOOK` which the consumer target could pass?

```swift
swiftSettings: [
    .define("NO_PLAYBOOK", .when(configuration: .release)),
]
```

**Comment by BarredEwe** [comment]
@markst Very cool and simple solution!

**Comment by BarredEwe** [comment]
> I suppose we may not want to wrap in DEBUG since some people may wish to create a distribution build with Playbook?

I think so, it is better to leave this feature configurable.

**Comment by BarredEwe** [comment]
@markst Can you please add information about this flag in Readme.md?

File: README.md [hunk 1]
```diff
@@ -238,6 +238,26 @@ playbook_configuration:
 - `imports` - Additional imports for the generated Playbook/Tests. Optional parameter.
 - `testable_imports` - Additional `@testable` imports for the generated Playbook/Tests. Optional parameter.
 
+Here's the updated copy for Prefire:
```
Comment (markst) [inline README.md:241]: 🙈

**Pull request merged**